src/components/implementation/simple_vmm/vmm/vcpuid.c

```diff
@@ -0,0 +1,763 @@
+#include <cos_types.h>
+#include <cos_debug.h>
+#include <stddef.h>
+#include <string.h>
+#include <vmrt.h>
+#include <vmx_msr.h>
+#include "cpu.h"
+#include "cpuid.h"
+#include "vcpuid.h"
+#include "cpu_caps.h"
+#include "apicreg.h"
+
+/*
+ * Copyright (C) 2018-2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+static struct cpuinfo_x86 boot_cpu_data;
+static struct vcpuid_entry vcpuid_entries[MAX_VM_VCPUID_ENTRIES];
+static u32_t g_entry_nr = 0, g_vcpuid_level, g_vcpuid_xlevel;
+void memcpy_erms(void *d, const void *s, size_t slen)
```

Comment:
Not great to have architecture-specific code here. But copying from the other project seems most expedient.